File: metadata/metadata-api/src/main/java/com/thinkbiganalytics/metadata/api/datasource/UserDatasource.java
package com.thinkbiganalytics.metadata.api.datasource;

import com.thinkbiganalytics.security.AccessControlled;

import java.util.Optional;

import javax.annotation.Nonnull;

/**
 * Defines a data source managed through Kylo by a user.
 */
public interface UserDatasource extends Datasource, AccessControlled {

    /**
     * Gets the additional properties for this data source.
     *
     * @return the details
     */
    @Nonnull
    Optional<? extends DatasourceDetails> getDetails();

    /**
     * Sets the name of this data source.
     *
     * @param name the name
     */
    void setName(@Nonnull String name);

    /**
     * Gets the type of this data source.
     *
     * @return the type
     */
    @Nonnull
    String getType();

    /**
     * Sets the type of this data source.
     *
     * @param type the type
     */
    void setType(@Nonnull String type);

}

File: ui/ui-app/src/main/resources/static/js/feed-mgr/model/user-datasource.ts
/**
 * A data source created and managed by a Kylo user.
 */
export interface UserDatasource {

    /**
     * Unique id for this data source.
     */
    id: string;

    /**
     * Human-readable name.
     */
    name: string;

    /**
     * Description of this data source.
     */
    description: string;

    /**
     * Type name of this data source.
     */
    type: string;
}
